basic_test.sh

```diff
@@ -0,0 +1,144 @@
+#!/bin/bash
+
+# Basic CI/CD test script for mini-gdbstub
+# This is a MVP (Minimum Viable Product) version
+
+set -e  # Exit on any error
+
+# Configuration
+GDBSTUB_HOST="127.0.0.1"
+GDBSTUB_PORT="1234"
+TIMEOUT=10
+GDB_TIMEOUT=30
+BUILD_DIR="build"
+
+# Colors for output
+RED='\033[0;31m'
+GREEN='\033[0;32m'
+YELLOW='\033[1;33m'
+NC='\033[0m' # No Color
+
+log_info() {
+    echo -e "${GREEN}[INFO]${NC} $1"
+}
+
+log_warn() {
+    echo -e "${YELLOW}[WARN]${NC} $1"
+}
+
+log_error() {
+    echo -e "${RED}[ERROR]${NC} $1"
+}
+
+# Cleanup function - will be called on script exit
+cleanup() {
+    log_info "Cleaning up..."
+    if [ ! -z "$GDBSTUB_PID" ]; then
+        log_info "Stopping gdbstub server (PID: $GDBSTUB_PID)"
+        kill $GDBSTUB_PID 2>/dev/null || true
+        wait $GDBSTUB_PID 2>/dev/null || true
+    fi
+    # Clean up temporary files
+    rm -f /tmp/gdb_test.gdb /tmp/gdb_output.log
+}
+
+# Set trap to call cleanup on script exit
+trap cleanup EXIT
+
+# Step 1: Build stage
+log_info "=== Step 1: Building project ==="
+make clean || true
+make build-emu || {
+    log_error "Build failed"
+    exit 1
+}
+log_info "Build completed successfully"
```

Comment:
I won't expect to build the project under the script, because this can make it inconvenient to check which step fails from the [dashboard](https://github.com/RinHizakura/mini-gdbstub/actions/runs/17525451352/job/49775219166).

Please check [.github/workflows/tests.yml](https://github.com/RinHizakura/mini-gdbstub/blob/main/.github/workflows/tests.yml). In fact, I already have one script to run some simple checks. Although it is quite simple, you can refer to this and merge these two for better coverage.